scheme/sevsnp/common.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2025 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package sevsnp
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/google/go-sev-guest/abi"
+	"github.com/google/go-sev-guest/kds"
+	"github.com/google/go-sev-guest/proto/sevsnp"
+	"github.com/veraison/cmw"
+	"github.com/veraison/corim/comid"
+	"github.com/veraison/corim/corim"
+	"github.com/veraison/ratsd/tokens"
+	"github.com/veraison/services/proto"
+)
+
+// measurementByUintKey looks up comid.Measurement in a CoMID by its MKey.
+//
+//	If no measurements are found, returns nil and no error. Otherwise,
+//	returns the error encountered.
+func measurementByUintKey(refVal comid.ValueTriple,
+	key uint64) (*comid.Measurement, error) {
+	for _, m := range refVal.Measurements.Values {
+		if m.Key == nil || !m.Key.IsSet() ||
+			m.Key.Type() != comid.UintType {
+			continue
+		}
+
+		k, err := m.Key.GetKeyUint()
+		if err != nil {
+			return nil, err
+		}
+
+		if k == key {
+			return &m, nil
+		}
+	}
+
+	return nil, nil
+}
+
+// comidFromJson accepts a CoRIM in JSON format and returns a single CoMID
+//
+//	Returns error if there are more than a single CoMID, or passes on
+//	error from corim routine.
+func comidFromJson(buf []byte) (*comid.Comid, error) {
+	extractedCorim, err := corim.UnmarshalUnsignedCorimFromJSON(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(extractedCorim.Tags) > 1 {
+		return nil, errors.New("too many tags")
+	}
+
+	extractedComid, err := corim.UnmarshalComidFromCBOR(
+		extractedCorim.Tags[0],
+		extractedCorim.Profile,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return extractedComid, nil
+}
+
+func parseCertificateChainFromEvidence(tsm *tokens.TSMReport) (*sevsnp.CertificateChain, error) {
+	var certTable abi.CertTable
+
+	if err := certTable.Unmarshal(tsm.AuxBlob); err != nil {
```

Comment:
Should we first make sure that AuxBlob is not nil?

Reply:
OK, I added this check. Thanks